File: character.rb
class Character
	def initialize()
		@posX = 1
		@posY = 1
	end
	
	def moveLeft()
		@posX -= 1
	end

	def moveRight()
		@posX += 1
	end

	def moveUp()
		@posY -= 1
	end

	def moveDown()
		@posY += 1
	end

	attr_reader :posX, :posY
end

File: battlesim/character.py
TILESIZE = 32
class Character(object):
    def __init__(self):
	    self.posX = 1 * TILESIZE
	    self.posY = 1 * TILESIZE

    def moveLeft(self):
	    self.posX -= 8

    def moveRight(self):
	    self.posX += 8

    def moveUp(self):
	    self.posY -= 8

    def moveDown(self):
	    self.posY += 8

    def snap(self):
        print self.posX, self.posY
        self.posX = (self.posX / TILESIZE + 1) * TILESIZE

        #self.posX = (self.posX / TILESIZE + 1) * TILESIZE
        #self.posY = (self.posY / TILESIZE) * TILESIZE
